test/src/test/java/org/corfudb/runtime/object/transactions/SnapshotTransactionContextTest.java

```diff
@@ -87,6 +90,22 @@ public void testSnapshotTxNestedExplicitTx() {
         }));
         t(0, this::TXEnd);
         t(0, this::TXEnd);
+    }
 
+
+    /**
+     * Ensure that TransactionAbortedException(AbortCause.UNSUPPORTED_OPERATION) is thrown when
+     * doing a mutation inside of a snapshot read.
+     */
+    @Test
+    public void testUnsupportedOperationException() {
+        try {
+            SnapshotTXBegin();
+            put("k" , "v1");
```

Comment:
I see what you mean but the function that you modified is TXExecuteInner that is called from TXExecute that is an "internal API" called for implicit transaction. This is used for SMRMaps methods annotated @Transactional.

What you tested here is the abortTransaction function that is called when the UNSUPPORTED_OPERATION is thrown during a non-nested transaction. This is part of the test. Before your fix, the transaction would also be aborted but with an UNDEFINED cause.

If you want to test that within a implicit transaction an UnsupportedOperationException will be thrown to the outer transaction without any retry, you have to use a SMRMap method annotated as @Transactional (e.g. putIfAbsent) so you can exercise TXExecuteInner.

Also, you might also want to test the retry logic within TXExecuteInner but this might be a bit harder to test.